### ISSUE TITLE
[Bug] alignment issue search engines view

### ISSUE THREAD

**Issue by cadeyrn** [mozilla-mobile/fenix#5236]
follow-up from #1442.

As I said in #1142:

> While it's definitively better for the search engine icons (still not perfectly aligned but it looks better now) the misalignment of the settings icon is still noticeable in my opinion.

@lime124 said:

> could you file a follow up issue for the settings alignment? i find it odd that didn't get fixed along with the search icons, but that was the intent. there will have to do some investigation and figure out why it's not happening. thanks!

I attach a screenshot to show the current state:

![Screenshot_20190911-221036](https://user-images.githubusercontent.com/1100614/64731382-647aa480-d4e1-11e9-81d1-9b9bdaf364de.png)

Let me know what you think. Maybe it's good enough and it's only me. But my feeling is that the label of the search engine settings link is a few pixel too far down.

**Comment by Brandon-Davison** [comment]
Could I pick this up? I'm looking for a good first issue to get myself involved in open source.

**Comment by neekless** [comment]
I don't think this is limited to the search engine settings label. Looking at the layout bounds it seems like all of the labels are a couple pixels too low.

![Screenshot_20190913-145237](https://user-images.githubusercontent.com/22426822/64843197-fa065900-d5f3-11e9-930d-1efdb78af340.jpg)

**Comment by olsenme** [comment]
Has this issue been assigned?

**Comment by ekager** [comment]
This is not ready to be worked on. Adding the label for UX Feedback. 
Note to engineering if this requires further work - These list item layouts are in AC.

**Comment by brampitoyo** [comment]
@cadeyrn What’s your proposed solution. Would you be able to mock it up?

Did you propose to:

* Reduce vertical space between each menu item, so they’re not so wide?
* Move each text a couple of pixels up, so they’re vertical centred?

I’d be happy to follow measurements specified by Material Design, especially if it’s trivial to do (which sounds like it’s not?). https://material.io/components/lists/

**Comment by darkwing** [comment]
@brampitoyo  Is this still an actionable issue?  Do you still see a problem with layout?

**Comment by brampitoyo** [comment]
@darkwing Yes. This issue is still actionable.

TL;DR – Generally, shifting all text elements inside labels up by 1dp will fix this vertical alignment issue. However, I’m also worried that we’ll introduce more bugs!

Explanation below. @cadeyrn, let me know if I’ve got it wrong.

When we show a label text alongside an icon, we position it so that its top and bottom paddings are equal.

<img width="360px" src="https://user-images.githubusercontent.com/51007/74391703-dcbb0c80-4e69-11ea-9178-672c483dab4f.png">

The problem: when these list items are arranged next to each other, the text – which otherwise have mathematically perfect spacing – sits optically too low.

<img width="360px" src="https://user-images.githubusercontent.com/51007/74391942-7a164080-4e6a-11ea-8a88-967f35e06cfa.png">

The fix that @cadeyrn proposed is similar to the “text baseline” concept (highlighted in “C” green colour), where the text doesn’t have equal top/bottom spacing, but is instead offset up by about 1dp.

Source: https://material.io/components/lists/#specs

<img width="760" src="https://user-images.githubusercontent.com/51007/74392096-eee97a80-4e6a-11ea-89ee-539a6c041c52.png">

In other words, I recommend **shifting our label text up by 1dp**. @cadeyrn what do you think?

**Comment by cadeyrn** [comment]
@brampitoyo Sounds good!
